File: agents/code-clarity-agent/examples/bad_code.py
# Bad example with intentional issues

def calc(x, y): # No docstring, poor names
    # check if y is greater than the threshold
    if y > 0.15: # Magic number
        return x * y
    return x

class DataProcessor:
    def __init__(self, data):
        self.data = data

    def process(self): # No docstring
        processed_data = []
        for item in self.data:
            res = calc(item, 0.2) # Magic number
            processed_data.append(res)
        return processed_data

def main():
    d = [1, 2, 3, 4, 5] # poor name
    dp = DataProcessor(d)
    result = dp.process()
    print(result)

if __name__ == "__main__":
    main()

File: agents/code-clarity-agent/examples/bad_code.js
// Bad example with intentional issues

function calc(x, y) { // No docstring, poor names
    // check if y is greater than the threshold
    if (y > 0.15) { // Magic number
        return x * y;
    }
    return x;
}

class DataProcessor {
    constructor(data) {
        this.data = data;
    }

    process() { // No docstring
        let processedData = [];
        for (const item of this.data) {
            let res = calc(item, 0.2); // Magic number
            processedData.push(res);
        }
        return processedData;
    }
}

function main() {
    let d = [1, 2, 3, 4, 5]; // poor name
    let dp = new DataProcessor(d);
    const result = dp.process();
    console.log(result);
}

main();
